static/js/pages/certificates/CertificateGenerate.tsx

```diff
@@ -0,0 +1,136 @@
+import CopyButton from "components/CopyButton";
+
+interface Certs {
+  crt: string;
+  pfx: string;
+}
+
+const CertificateGenerate: FC = () => {
+  const [isGenerating, setGenerating] = useState(false);
+  const [certs, setCerts] = useState<Certs | null>(null);
+
+  const createCert = () => {
+    setGenerating(true);
+    // using timeout to avoid compute heavy generation in the main ui thread
+    setTimeout(() => {
+      const certs = generateCert();
+      setCerts(certs);
+      setGenerating(false);
+    }, 10);
+  };
+
+  const downloadBase64 = (name: string, base64: string) => {
+    const linkSource = `data:application/octet-stream;base64,${base64}`;
+    const downloadLink = document.createElement("a");
+
+    downloadLink.href = linkSource;
+    downloadLink.download = name;
+    downloadLink.click();
+  };
+
+  const downloadText = (name: string, text: string) => {
+    const data = encodeURIComponent(text);
+    const linkSource = `data:text/plain;charset=utf-8,${data}`;
+    const downloadLink = document.createElement("a");
+
+    downloadLink.href = linkSource;
+    downloadLink.download = name;
+    downloadLink.click();
+  };
+
+  return (
+    <BaseLayout title="Certificate generation">
+      <Row className="p-certificate-generate">
+        <ol className="p-stepped-list">
+          <li className="p-stepped-list__item">
+            <h3 className="p-stepped-list__title">Generate</h3>
+            <div className="p-stepped-list__content">
+              <Row>
+                <Col size={8}>
+                  <p>Click generate to create a new certificate</p>
+                </Col>
+                <Col size={4}>
+                  <Button
+                    onClick={createCert}
+                    appearance="positive"
+                    disabled={isGenerating || certs !== null}
+                    hasIcon={isGenerating}
+                  >
+                    {isGenerating && (
+                      <i className="p-icon--spinner is-light u-animation--spin"></i>
+                    )}
+                    <span>{isGenerating ? "Generating" : "Generate"}</span>
+                  </Button>
+                </Col>
+              </Row>
+            </div>
+          </li>
+          <li className="p-stepped-list__item">
+            <h3 className="p-stepped-list__title">Trust</h3>
+            <div className="p-stepped-list__content">
+              <Row>
+                <Col size={8}>
+                  Download <code>lxd-ui.crt</code> and add it to the LXD trust
+                  store
+                  <div className="p-code-snippet">
+                    <pre className="p-code-snippet__block--icon">
+                      <code>lxc config trust add Downloads/lxd-ui.crt</code>{" "}
+                      <CopyButton
+                        text="lxc config trust add Downloads/lxd-ui.crt"
+                        tooltipPosition="right"
+                      />
```

Comment:
As mentioned, this would probably better right-aligned.

Reply:
It is just the buttons to download and copy would be too close to each other.